src/cascadia/TerminalApp/TabBase.cpp

```diff
@@ -590,4 +586,49 @@ namespace winrt::TerminalApp::implementation
             VisualStateManager::GoToState(item, L"Normal", true);
         }
     }
+
+    TabCloseButtonVisibility TabBase::CloseButtonVisibility()
+    {
+        return _closeButtonVisibility;
+    }
+
+    void TabBase::CloseButtonVisibility(TabCloseButtonVisibility visibility)
+    {
+        _closeButtonVisibility = visibility;
+        _updateIsClosable();
+    }
+
+    void TabBase::_updateIsClosable()
+    {
+        if (ReadOnly())
+        {
+            TabViewItem().IsClosable(false);
+        }
+        else
+        {
+            switch (_closeButtonVisibility)
+            {
+            case TabCloseButtonVisibility::Never:
+                TabViewItem().IsClosable(false);
+                break;
+            case TabCloseButtonVisibility::Hover:
+                TabViewItem().IsClosable(true);
+                break;
+            case TabCloseButtonVisibility::ActiveOnly:
+            {
+                TabViewItem().IsClosable(_focused());
+                break;
+            }
+            default:
+                TabViewItem().IsClosable(true);

```

Comment:
We will never leave this function without calling `TabViewItem().IsClosable()` at least once. Can you factor out the common subexpression? No point in emitting 10 function calls when 2 will do :smile: